packages/legend-application-repl/src/stores/dataCube/DataCubeStore.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { GenericLegendApplicationStore } from '@finos/legend-application';
+import type { REPLServerClient } from '../../server/REPLServerClient.js';
+import { DataCubeEngine } from './DataCubeEngine.js';
+import { DataCubeState } from './DataCubeState.js';
+import { DisplayState, LayoutManagerState } from '../LayoutManagerState.js';
+import { DocumentationPanel } from '../../components/repl/DocumentationPanel.js';
+import { SettingsPanel } from '../../components/repl/SettingsPanel.js';
+
+export class DataCubeStore {
+  readonly applicationStore: GenericLegendApplicationStore;
+  readonly engine: DataCubeEngine;
+  readonly dataCubeState: DataCubeState;
```

Comment:
let's keep the comment about we might have multiple of this and move it down below. Thanks!